sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -93,6 +93,24 @@ object IntervalUtils {
 
   private val yearMonthPattern = "^([+|-])?(\\d+)-(\\d+)$".r
 
+  def safeFromYearMonthString(input: UTF8String): Option[Int] = {
+    try {
+      if (input == null || input.toString == null) {
+        throw new IllegalArgumentException("Interval year-month string must be not null")
+      } else {
+        val regex = "INTERVAL '([-|+]?[0-9]+-[-|+]?[0-9]+)' YEAR TO MONTH".r
```

Comment:
The keywords `INTERVAL` and `YEAR TO MONTH` are not mandatory by SQL standard. We should be able to parse only payload too.

Reply:
Also could you move the regexp out of the method. So, we can avoid its "compilation" per every call.

Reply:
Regarding to the pattern `'([-|+]?[0-9]+-[-|+]?[0-9]+)'`, should we handle the second sign before month? Take a look at the val above `yearMonthPattern`

Reply:
> Regarding to the pattern `'([-|+]?[0-9]+-[-|+]?[0-9]+)'`, should we handle the second sign before month? Take a look at the val above `yearMonthPattern`

How about current

Reply:
Can you help to list the case we need to support here,
```
INTERVAL [-|+]'[+|-]0-0' YEAR TO MONTH
[+|-]0-0
```
Any more case need to support here?

Reply:
First of all, we must support our own output (in case insensitive manner by default but ideally we should respect the SQL config `spark.sql.caseSensitive`):
1. ANSI_STYLE, like
INTERVAL -'-10-1' YEAR TO MONTH
2. HIVE_STYLE like
10-1 or -10-1

Rules from the SQL standard:
```
<interval literal> ::=
  INTERVAL [ <sign> ] <interval string> <interval qualifier>
<interval string> ::=
  <quote> <unquoted interval string> <quote>
<unquoted interval string> ::=
  [ <sign> ] { <year-month literal> | <day-time literal> }
<year-month literal> ::=
  <years value> [ <minus sign> <months value> ]
  | <months value>
<years value> ::=
  <datetime value>
<months value> ::=
  <datetime value>
<datetime value> ::=
  <unsigned integer>
<unsigned integer> ::= <digit>...
```

Reply:
> but ideally we should respect the SQL config `spark.sql.caseSensitive`):

Spark Catalyst parser not respect `spark.sql.caseSensitive` too, should we respect it here?

Reply:
For parsing I think it's OK to be always case insensitive. The Spark SQL parser is case insensitive.

Reply:
Ok, how about current approach